libs/dh/dev-examples/feature-processes/src/details/details.component.ts

```diff
@@ -0,0 +1,212 @@
+            />
+          }
+
+          @if (energyTimeSeriesRequest) {
+            <watt-description-list-item
+              [label]="t('details.meteringPointType')"
+              [value]="t('meteringPointType.' + energyTimeSeriesRequest.meteringPointType)"
+            />
+          }
+
+          @if (wholesaleRequest) {
+            <watt-description-list-item
+              [label]="t('details.priceType')"
+              [value]="t('priceType.' + wholesaleRequest.priceType)"
+            />
+          }
+        </watt-description-list>
+      </watt-drawer-heading>
+      <watt-drawer-content>
+        <dh-result [hasError]="hasError()" [loading]="loading()">
+          <vater-flex fill="vertical" gap="l" offset="l">
+            @if (process) {
+              <watt-progress-tracker>
+                @for (step of process.steps; track step; let i = $index) {
+                  <watt-progress-tracker-step
+                    [label]="'shared.steps.' + i + '.pending' | transloco"
+                    [status]="step.state"
+                    [current]="step.isCurrent"
+                  >
+                    {{ 'shared.steps.' + i + '.succeeded' | transloco }}
+                  </watt-progress-tracker-step>
+                }
+              </watt-progress-tracker>
+            }
+            <vater-flex scrollable fill="vertical" grow="0">
+              @if (calculation) {
+                <dh-calculation-details-grid-areas [gridAreas]="calculation.gridAreas" />
+              }
+            </vater-flex>
+          </vater-flex>
+        </dh-result>
+      </watt-drawer-content>
+    </watt-drawer>
+  `,
+})
+export class DhProcessDetailsComponent {
+  private navigation = inject(DhNavigationService);
+  private processQuery = lazyQuery(GetProcessDocument);
+
+  loading = this.processQuery.loading;
+  hasError = this.processQuery.hasError;
+  result = computed(() => this.processQuery.data()?.processById);
+  startedAtOrScheduledAt = computed(() => this.result()?.scheduledAt ?? this.result()?.scheduledAt);
+
+  calculationDetails = computed(() =>
+    this.result() && this.result()?.__typename === 'Calculation'
+      ? (this.result() as DhProcessCalculation)
+      : null
+  );
+
+  energyTimeSeriesRequestDetails = computed(() =>
+    this.result() && this.result()?.__typename === 'RequestCalculatedEnergyTimeSeriesResult'
+      ? (this.result() as DhProcessEnergyTimeSeriesRequest)
+      : null
+  );
+
+  wholesaleRequestDetails = computed(() =>
+    this.result()?.__typename === 'RequestCalculatedWholesaleServicesResult'
+      ? (this.result() as DhProcessWholesaleRequest)
+      : null
+  );
+
+  // Param value
+  id = input.required<string>();
+  drawer = viewChild(WattDrawerComponent);
+
+  constructor() {
+    afterRenderEffect(() => {
+      this.processQuery.query({ variables: { id: this.id() } });
+      this.drawer()?.open();
+    });
```

Comment:
suggestion: is there a reason to query here in `afterRenderEffect` as opposed to just using a regular `query`? and if there is, I think it would be a good idea to add a comment here.

it could also be cool to have an "autoOpen" input or similar on the drawer instead of having to do these shenanigans

Reply:
The query is in an effect because its needs the id to be set. As to why it is a afterRenderEffect instead of an normal effect is because it saw issues with values changing after render and we get that angular error https://angular.dev/errors/NG0100

Think its a good idea with auto open

Reply:
Drawer has some odd logic around reusing the existing drawer if a new one is opened. The auto open input, breaks this logic.
Might do a PR later to see if it can be implemented.

Reply:
I really need to update `util-apollo` to accept a variables function just like `request` in the Angular resource API - that should prevent this exact issue